contracts/vaults/RibbonDeltaVault.sol

```diff
@@ -233,5 +280,46 @@ contract RibbonDeltaVault is RibbonVault, OptionsDeltaVaultStorage {
             counterpartyThetaVault.optionAuctionID(),
             orders
         );
+
+        _updatePPS();
+    }
+
+    /**
+     * @notice Updates the price per share of the current round. The current round
+     * pps will change right after call rollToNextOption as the gnosis auction contract
+     * takes custody of a % of `asset` tokens, and right after we claim the tokens from
+     * the action as we may recieve some of `asset` tokens back alongside the oToken,
+     * depending on the gnosis auction outcome. Finally it will change at the end of the week
+     * if the oTokens are ITM
+     */
+    function _updatePPS() private {
+        (, uint256 newPricePerShare, ) =
+            VaultLifecycle.rollover(totalSupply(), vaultParams, vaultState);
```

Comment:
There's a lot of extra steps that we can avoid in the `rollover` function, it's worth it to just copy paste the newPricePerShare calculation into a new helper function

Reply:
agree